lib/resources/package.rb

```diff
@@ -71,13 +73,27 @@ def version
     def to_s
       "System Package #{@package_name}"
     end
+
+    private
+
+    def evaluate_missing_requirements
```

Comment:
While I agree that we need to find a pattern for checking missing options, the current approach feels bloated and very complicated. I'd like to replace this with a version that is more lightweight and more flexible to be used in all resources. I do not have an idea right now, therefore this is not a PR blocker.